In [1]:
import pandas as pd


In [2]:
df = pd.read_csv(r"C:\Users\DELL\Downloads\project1_df.csv")

In [3]:
df.head(2)

,CID,TID,Gender,Age Group,Purchase Date,Product Category,Discount Availed,Discount Name,Discount Amount (INR),Gross Amount,Net Amount,Purchase Method,Location
0,943146,5876328741,Female,25-45,30/08/2023 20:27:08,Electronics,Yes,FESTIVE50,64.30,725.304000,661.004000,Credit Card,Ahmedabad
1,180079,1018503182,Male,25-45,23/02/2024 09:33:46,Electronics,Yes,SEASONALOFFER21,175.19,4638.991875,4463.801875,Credit Card,Bangalore


In [4]:
df.nunique()

CID                      29071
TID                      55000
Gender                       3
Age Group                    5
Purchase Date            54988
Product Category             9
Discount Availed             2
Discount Name                5
Discount Amount (INR)    20578
Gross Amount             54807
Net Amount               54936
Purchase Method              8
Location                    14
dtype: int64

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 55000 entries, 0 to 54999
Data columns (total 13 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   CID                    55000 non-null  int64  
 1   TID                    55000 non-null  int64  
 2   Gender                 55000 non-null  object 
 3   Age Group              55000 non-null  object 
 4   Purchase Date          55000 non-null  object 
 5   Product Category       55000 non-null  object 
 6   Discount Availed       55000 non-null  object 
 7   Discount Name          27415 non-null  object 
 8   Discount Amount (INR)  55000 non-null  float64
 9   Gross Amount           55000 non-null  float64
 10  Net Amount             55000 non-null  float64
 11  Purchase Method        55000 non-null  object 
 12  Location               55000 non-null  object 
dtypes: float64(3), int64(2), object(8)
memory usage: 5.5+ MB


In [6]:
df['Purchase Date'] = pd.to_datetime(df['Purchase Date'], format='%d/%m/%Y %H:%M:%S')

In [7]:
print(df['Purchase Date'].head())
print(df['Purchase Date'].dtype)

0   2023-08-30 20:27:08
1   2024-02-23 09:33:46
2   2022-03-06 09:09:50
3   2020-11-04 04:41:57
4   2022-05-31 17:00:32
Name: Purchase Date, dtype: datetime64[ns]
datetime64[ns]


In [8]:
df["Age Group"].value_counts()


Age Group
25-45           22010
18-25           16431
45-60           11104
under 18         2771
60 and above     2684
Name: count, dtype: int64

In [9]:
df["Product Category"].value_counts()

Product Category
Electronics          16574
Clothing             10968
Beauty and Health     8332
Sports & Fitness      5557
Home & Kitchen        5489
Books                 2762
Other                 2171
Pet Care              1618
Toys & Games          1529
Name: count, dtype: int64

## Hypotheses Test
### Objective
- How Product category impact sales
### Hypotheses
- Null Hypothesis(Ho)- Product category does not affect sales
- Alternate Hypothesis(H1)- Product category  affects sales

In [10]:
categories = df['Product Category'].unique()
nets = [df[df['Product Category'] == i]['Net Amount'] for i in categories]

In [11]:
from scipy.stats import f_oneway
f_stat, p_val = f_oneway(*nets)

print(f"F-statistic: {f_stat}")
print(f"P-value: {p_val}")

F-statistic: 1.2149529696647237
P-value: 0.2852830641165648


In [12]:
alpha = 0.05
if p_val < alpha:
    print("Reject Null Hypothesis  -> Product category significantly affects sales")
else:
    print("Fail to Reject Null Hypothesis  -> Product category does not significantly affect sales")

Fail to Reject Null Hypothesis  -> Product category does not significantly affect sales


### Insights 
- Statistical testing shows no significant difference in average Net Amount across categories (p > 0.05).
- However, total revenue varies significantly due to different transaction volumes, with Electronics and Clothing contributing the most.

In [13]:
df["Discount Availed"].value_counts()

Discount Availed
No     27585
Yes    27415
Name: count, dtype: int64

## Hypotheses Test
### Objective
- How does discount impact sales
### Hypotheses
- Null Hypothesis(Ho)- Discount Does not Impact sales
- Alternate Hypothesis(H1)- Discount  Impacts sales

In [14]:
Discounted=df[df["Discount Availed"]=="Yes"]["Net Amount"]
Not_Discounted=df[df["Discount Availed"]=="No"]["Net Amount"]

In [15]:
from scipy.stats import ttest_ind
t_stat,p_value=ttest_ind(Discounted,Not_Discounted)
print("p_value:",p_value)

p_value: 1.6213304988087257e-78


In [16]:
if p_value< 0.05:
    print(" reject Null Hypothesis(Ho): Discount Impacts sales ")
else:
     print(" fail to reject Null Hypothesis(Ho): Discount Does not Impact sales")

 reject Null Hypothesis(Ho): Discount Impacts sales 


In [17]:
d = {"Discount Amount (INR)": "mean",  "Gross Amount": "sum","CID": "count"}
result =df.groupby(["Product Category","Discount Availed"]).agg(d)
result = result.reset_index()
result['Gross Amount'] = result['Gross Amount'].astype(int)
print(result)

     Product Category Discount Availed  Discount Amount (INR)  Gross Amount  \
0   Beauty and Health               No               0.000000      12850343   
1   Beauty and Health              Yes             277.546323      12469785   
2               Books               No               0.000000       4167437   
3               Books              Yes             273.104593       4144707   
4            Clothing               No               0.000000      16224498   
5            Clothing              Yes             273.410331      16507563   
6         Electronics               No               0.000000      25101064   
7         Electronics              Yes             275.489072      24642442   
8      Home & Kitchen               No               0.000000       8201259   
9      Home & Kitchen              Yes             273.421925       8452540   
10              Other               No               0.000000       3246552   
11              Other              Yes             2

### Insights 
- High-revenue categories are Electronics, Clothing, and Beauty & Health, contributing most to total sales.
- Discounts slightly increase revenue for mid-range categories like Clothing and Home & Kitchen.
- High-ticket categories like Electronics and Pet Care generate slightly more revenue without discounts.
- Average discount is around 270–280 INR and consistent across all categories.
### Recommendations 
- Focus marketing and inventory on high-revenue categories to maximize overall sales.
- Offer discounts selectively for mid-range or impulse-buy categories to boost transactions.
- Reduce or optimize discounts for high-value categories to protect margins.
- Monitor revenue vs discount effectiveness regularly to refine pricing and promotion strategies.

In [18]:
df["Location"].value_counts()

Location
Mumbai       11197
Delhi        10799
Bangalore     8249
Hyderabad     5545
Chennai       4368
Pune          3781
Ahmedabad     2785
Kolkata       2709
Jaipur        1678
Lucknow       1136
Other         1046
Varanasi       606
Dehradun       553
Srinagar       548
Name: count, dtype: int64

## Hypotheses Test
### Objective
- How Location impact sales
### Hypotheses
- Null Hypothesis(Ho)- location does not affect sales
- Alternate Hypothesis(H1)- Location  affects sales

In [19]:
locations = df['Location'].unique()
net_amount_per_location = [df[df['Location'] == loc]['Net Amount'] for loc in locations]
f_stat, p_val = f_oneway(*net_amount_per_location)

print(f"F-statistic: {f_stat}")
print(f"P-value: {p_val}")

F-statistic: 0.4591697188741771
P-value: 0.9472566781616057


In [20]:
alpha = 0.05
if p_val < alpha:
    print("Reject Null Hypothesis  -> Location affects sales ")
else:
    print("Fail to Reject Null Hypothesis  -> location does not affect sales")

Fail to Reject Null Hypothesis  -> location does not affect sales


In [21]:
d={"Net Amount":"sum","TID":"count"}
amount=df.groupby("Location").agg(d)
amount = amount.reset_index()
amount["Net Amount"] = amount["Net Amount"].astype(int)
print(amount)

     Location  Net Amount    TID
0   Ahmedabad     8114984   2785
1   Bangalore    23619545   8249
2     Chennai    12635181   4368
3    Dehradun     1569218    553
4       Delhi    31098974  10799
5   Hyderabad    15813386   5545
6      Jaipur     4854492   1678
7     Kolkata     7805354   2709
8     Lucknow     3270708   1136
9      Mumbai    32083841  11197
10      Other     3079421   1046
11       Pune    10902138   3781
12   Srinagar     1585926    548
13   Varanasi     1744065    606


### Insights 
- Location is not impacting sales
- Focus on other aspect
- Average order value is similar across locations (no statistical difference),
but total revenue differs significantly due to higher transaction volume in metro cities

In [22]:
df["Age Group"].value_counts()

Age Group
25-45           22010
18-25           16431
45-60           11104
under 18         2771
60 and above     2684
Name: count, dtype: int64

## Hypotheses Test
### Objective
- How Age Group impact sales
### Hypotheses
- Null Hypothesis(Ho)- Age Group does not affect sales
- Alternate Hypothesis(H1)-Age Group affects sales

In [23]:
Age_Group = df['Age Group'].unique()
net_amount_per_age_group = [df[df['Age Group'] == loc]['Net Amount'] for loc in Age_Group]
f_stat, p_val = f_oneway(*net_amount_per_age_group)

print(f"F-statistic: {f_stat}")
print(f"P-value: {p_val}")

F-statistic: 3.1093649956037326
P-value: 0.014385382945885266


In [24]:
alpha = 0.05
if p_val < alpha:
    print("Reject Null Hypothesis  -> Age Group affects sales ")
else:
    print("Fail to Reject Null Hypothesis  -> Age Group does not affect sales")

Reject Null Hypothesis  -> Age Group affects sales 


In [25]:
d = {"Net Amount": ["mean", "sum"],"CID": "count"}
age=df.groupby("Age Group").agg(d)
age["Net Amount"] = age["Net Amount"].astype(int)
print(age)

             Net Amount              CID
                   mean       sum  count
Age Group                               
18-25              2848  46804916  16431
25-45              2904  63933310  22010
45-60              2853  31683799  11104
60 and above       2886   7747872   2684
under 18           2889   8007341   2771


### Insights

- Age 25–45 contributes the highest total revenue (63,933,310) and has the highest number of customers (22,010).

- Younger (18–25) and middle-aged (45–60) groups have moderate revenue, while under 18 and 60+ contribute the least.

### Recommendations

- Focus marketing and promotions on the 25–45 age group, as they drive the most revenue and transactions.

- Targeted campaigns for 18–25 and 45–60 groups could increase revenue, while special offers may help engage under 18 and 60+ segments.

In [26]:
from sqlalchemy import create_engine


user = 'root'
password = 'YPUR PASSWORD' ### YOUR SQL PASSWORD
host = 'localhost'  
port = 3306
database = 'ecommerce'


engine = create_engine(f'mysql+mysqlconnector://{user}:{password}@{host}:{port}/{database}')

In [27]:
df.to_sql("ecommerce_orders", engine, if_exists="replace", index=False)

print("Data uploaded successfully!")

Data uploaded successfully!
